### ISSUE TITLE
Problem generating Diffreport after XCMSnEXp to xcmsSet object conversion

### ISSUE THREAD

**Issue by CLechaplais** [sneumann/xcms#367]
Hi Steffen, as you suggested to me, here is my problem :
> Since the latest version of XCMS (version 3.4.2) seems to be more 
> appropriate for dealing with large datasets, we would like to use it
> on 
> our current project. Unfortunately, we have problems in creating the 
> diffreport after the conversion of the XCMSnExp object into a
> xcmsSet 
> object (function >as() ). On the one hand we lose the sample names
> and 
> on the other hand the group notion (eg. KO,WT) used for the
> statistics, 
> leading to an uncomplete diffreport (with missing statistics and
> sample 
> names) and EIC with only 2 samples with that XCMS warning message
> “Too 
> few samples per class, skipping t-test.”.
>
> To understand better, we’ve tried following these 2 vignettes :
> -           LCMS data preprocessing and analysis with xcms
>
https://bioconductor.org/packages/release/bioc/vignettes/xcms/inst/doc/xcms.html
> -          MTBLS2 Processing and Analysis with xcms and CAMERA and 
> export to MetaboLights
>
https://bioconductor.org/packages/release/data/experiment/vignettes/mtbls2/inst/doc/MTBLS2-xcms3.html
>
>
> When we use the 1st one, everything is OK until we convert XCMSnExp 
> object for creating the diffreport. In this case, the diffreport
> looks 
> like the one we create on our own dataset.
> On the opposite, when we use the 2nd one, everything works fine; we
> get 
> the full statistics, eic and diffreport.
>
> Could you please indicate us how to run XCMS with data from the
> first 
> vignette, including the creation of a full diffreport?

Thanks for help

**Comment by sneumann** [comment]
Hi, just trying to reproduce your issue here, will keep you posted.

**Comment by sneumann** [comment]
So, just for the archive: the easiest way to reproduce is running 
```
docker run --rm -it -p 127.0.0.1:8787:8787 -e DISABLE_AUTH=true bioconductor/release_metabolomics2`
```
which currently has `xcms_3.4.4` installed. It is available through http://127.0.0.1:8787/

You'll need mtbls2 and the BiocStyle for the vignette:
```
BiocManager::install("mtbls2")
BiocManager::install("BiocStyle")
```
After running all chunks in the xcms.Rmd, data is in `xdata`,
which can be converted back to an xcms1 object:
```
> xset <- as(xdata, "xcmsSet")
Note: you might want to set/adjust the 'sampclass' of the returned xcmSet object before proceeding with the analysis.
> xset
An "xcmsSet" object with 12 samples

Time range: 2496.3-4230.4 seconds (41.6-70.5 minutes)
Mass range: 200.1-599.4 m/z
Peaks: 6460 (about 538 per sample)
Peak Groups: 351 
Sample classes: ko15.KO, ko16.KO, ko18.KO, ko19.KO, ko21.KO, ko22.KO, wt15.WT, wt16.WT, wt18.WT, wt19.WT, wt21.WT, wt22.WT 

Feature detection:
 o Peak picking performed on MS1.
 o Scan range limited to  1 - 1278 
Profile settings: method = bin
                  step = 0.1

Memory usage: 3.45 MB
```
sample names and as the note implies, sample class information and is incorrect. 
That can be fixed via the following, and then the diffreport can be generated:
```
sampnames(xset) <- pData(xdata)$sample_name
sampclass(xset) <- pData(xdata)$sample_group

xset2 <- fillPeaks(xset)
dr <- diffreport(xset2)
```
Hope that helped, 
yours, Steffen

**Comment by CLechaplais** [comment]
Hi Steffen,
I have both tried on the faahKO and our own datasets, everything is now ok.
Thanks much for your great and fast answer.
Kind regards,
Christophe
